### DISCUSSION TITLE
Direct IP vs ID performance

### DISCUSSION THREAD

**Discussion by louwe** [rustdesk/rustdesk#4380] [Q&A] [answered]
I only use Rust Desk over secure LAN or tunneled over Wireguard so I'm not worried about security and just want the best and fastest performance. In this case, would it be better to just use direct IP always or is the performance hit on ID negligible and makes no difference?

Also, seems like the connection needs to be direct IP and my custom LAN DNS names don't seem to work. Am I doing something wrong?

**Comment by SergeyMy** [comment]
use direct IP = disables encryption!
the performance hit for the identifier is negligible! but allows the use of unique IDs
on your ID-server, you can arbitrarily change (allow change) ID

**Reply by louwe** [reply]
I'm not worried about encryption as I already stated, it's always used over a trusted LAN or tunneled over Wireguard so encryption is not a concern. I just wanted an authoritative answer that it is better and worth the effort of setting up.

Especially if it also means that it will improve latency because the internal traffic stays internal and doesn't leak out to the internet (in the case of two machines in the LAN). Also, my LAN bandwidth is 1G symmetric while my internet is only 350M down and 11M up.

**Reply by SergeyMy** [reply, marked as answer]
1. you need your own server! (exclude external traffic)
https://rustdesk.com/server/
2. How does RustDesk work
https://github.com/rustdesk/rustdesk/wiki/How-does-RustDesk-work%3F

**Reply by louwe** [reply]
Ah thank you for that sequence diagram! That's what I was looking for.
